google/cloud/bigquery/job.py

```diff
@@ -2067,14 +2067,20 @@ def destination_uri_file_counts(self):
     def to_api_repr(self):
         """Generate a resource for :meth:`_begin`."""
 
+        configuration = self._configuration.to_api_repr()
         source_ref = {
             "projectId": self.source.project,
             "datasetId": self.source.dataset_id,
-            "tableId": self.source.table_id,
         }
 
-        configuration = self._configuration.to_api_repr()
-        _helpers._set_sub_prop(configuration, ["extract", "sourceTable"], source_ref)
+        if isinstance(self.source, TableReference):
+            source_ref["tableId"] = self.source.table_id
+            source = "sourceTable"
+        else:
+            source_ref["modelId"] = self.source.model_id
+            source = "sourceModel"
```

Comment:
One can see a potential problem here, when `self.source` is neither, `TableReference` nor `ModelRefrence`, e.g. `None` or some other type..

Reply:
Yes i understand , but it's `if else` statement so by default it take `sourceModel` as it set in else statement

Reply:
And if for some reason, such as another code change, the `source` is neither, model nor table, what then?

Reply:
Yes , agree for that need to rewrite as per scenario as can't use if elif statement for long.
For now can i make sourceTable as a default and reset value for model?